contracts/RewardPool.sol

```diff
@@ -256,6 +289,179 @@ contract RewardPool is AccessControl, ReentrancyGuard {
     emit Withdraw(amount);
   }
 
+  /**
+   * @dev Internal function to get the period id
+   * @param periodStart Start of the reward period (unix timestamp)
+   * @param periodEndExclusive End of the reward period (unix timestamp)
+   * @return bytes32 period id
+   */
+  function _getPeriodId(
+    uint48 periodStart,
+    uint48 periodEndExclusive
+  ) internal pure returns (bytes32) {
+    if (periodStart >= periodEndExclusive)
+      revert PeriodInvalid(periodStart, periodEndExclusive);
+
+    return keccak256(abi.encode(periodStart, periodEndExclusive));
+  }
+
+  /**
+   * @dev Allows the owner to update kpis for a reward period
+   * @param kpis Array of kpis to add
+   * @param periodStart Start of the reward period (unix timestamp)
+   * @param periodEndExclusive End of the reward period (unix timestamp)
+   * @param kpiFunctionId Bytes32 identifier of the kpi function (e.g., github commit hash)
+   * @notice Allowed only for address with DEFAULT_ADMIN_ROLE
+   */
+  function updatePeriodKpis(
+    IRewardFunction.Kpi[] calldata kpis,
+    uint48 periodStart,
+    uint48 periodEndExclusive,
+    bytes32 kpiFunctionId
+  ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    bytes32 periodId = _getPeriodId(periodStart, periodEndExclusive);
+
+    PeriodData storage period = _periods[periodId];
+
+    if (period.processedAt != 0) revert PeriodAlreadyProcessed(periodId);
+
+    for (uint256 i = 0; i < kpis.length; i++) {
+      if (kpis[i].referrerAddress == address(0))
```

Comment:
should we rename `referrerAddress` to `user` in `IRewardFunction` to be consistent with addRewards? Although I wonder if that will be confusing because `user` has a different meaning in our scripts.

Reply:
Yes I was wondering about this naming too. This contract uses "user" for the referrer. But I guess technically "referrer" is more precised, right?
Would be good to stay consistent and avoid any confusion.

Reply:
Hmm, I think my vote would be for `referrer` here in place of `user` -- I think it's easy to get confused and think that `user` is supposed to mean "end user" or "referred user" address instead of "builder/referrer address". Not a huge deal either way, I suppose.